frontend/lib/create_listing/view/create_listing.view.dart

```diff
@@ -0,0 +1,414 @@
+  }
+
+  // Header widgets
+  final appBarTitle = const Text(
+    'Create Post',
+    style: TextStyle(
+      fontSize: 32,
+      fontWeight: FontWeight.bold,
+    ),
+  );
+
+  final postButton = const Text(
+    'Post',
+    style: TextStyle(
+      color: Color(0xFF2F9CCF),
+      fontSize: 16,
+    ),
+  );
+
+  // Form widgets
+  Widget _buildImageSelector() {
+    return Consumer<CreateListingViewModel>(
+      builder: (_, model, __) {
+        final bool showError = !model.hasImage && _showValidationErrors;
+        
+        return GestureDetector(
+          onTap: () async {
+            // Your existing image selection logic
+          },
+          child: Container(
+            height: 300,
+            decoration: BoxDecoration(
+              color: Colors.grey[200],
+              borderRadius: BorderRadius.circular(12),
+              border: Border.all(
+                color: showError
+                    ? const Color(0xFFFF5252)
+                    : Colors.grey[300]!,
+                width: 1,
+              ),
+            ),
+            child: model.image != null
+                ? ClipRRect(
+                    borderRadius: BorderRadius.circular(12),
+                    child: Image.file(
+                      model.image!,
+                      fit: BoxFit.cover,
+                      width: double.infinity,
+                    ),
+                  )
+                : Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      Icon(
+                        Icons.add_photo_alternate_outlined,
+                        size: 48,
+                        color: showError
+                            ? const Color(0xFFFF5252)
+                            : Colors.grey[600],
+                      ),
+                      const SizedBox(height: 8),
+                      Text(
+                        'Add photos / video',
+                        style: TextStyle(
+                          color: showError
+                              ? const Color(0xFFFF5252)
+                              : Colors.grey[600],
+                          fontSize: 16,
+                        ),
+                      ),
+                      if (showError)
+                        const Padding(
+                          padding: EdgeInsets.only(top: 4),
+                          child: Text(
+                            'Add a photo to continue.',
+                            style: TextStyle(
+                              color: Color(0xFFFF5252),
+                              fontSize: 12,
+                            ),
+                          ),
+                        ),
+                    ],
+                  ),
+          ),
+        );
+      },
+    );
+  }
+
+  Widget _buildTextField(String label, TextEditingController controller, {bool isPrice = false}) {
```

Comment:
Is this component going to be shared with other pages? If yes, we should keep it in a shared folder so that other pages can access it. Regardless, as above, we should create a widget for this at least in the components folder for this feature.